source/server/filter_chain_manager_impl.h

```diff
@@ -132,6 +190,8 @@ class FilterChainManagerImpl : public Network::FilterChainManager,
   // and application protocols, using structures defined above.
   DestinationPortsMap destination_ports_map_;
   const Network::Address::InstanceConstSharedPtr address_;
+  Configuration::FactoryContext& parent_context_;
+  std::list<std::shared_ptr<Configuration::FilterChainFactoryContext>> factory_contexts_;
```

Comment:
This can be a unique_ptr for now. We can change to a shared_ptr later when it's more clear why we need a shared_ptr.

Reply:
Nice catch! Updated